### PR TITLE
Add examples of user inventory plugins

### PR DESCRIPTION
**Pull request by AlanCoding**

### CODE REVIEW

File: inventories/user_plugins/inventory_plugins/fox.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+EXAMPLES = r'''
+    # plugin: fox
+'''
+
+from ansible.plugins.inventory import BaseInventoryPlugin
+
+
+def ancient_mystery():
+    raise Exception('Gering-ding-ding-ding-dingeringeding')
```
Comment (AlanCoding) [inline inventories/user_plugins/inventory_plugins/fox.py]: @kdelee this serves a very specific testing purpose.

```
 [WARNING]:  * Failed to parse /Users/alancoding/Documents/repos/jlaska-ansible-playbooks/inventories/user_plugins/fox.yaml with auto plugin: Gering-ding-ding-ding-dingeringeding

  File "/Users/alancoding/Documents/repos/ansible/lib/ansible/inventory/manager.py", line 272, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/Users/alancoding/Documents/repos/ansible/lib/ansible/plugins/inventory/auto.py", line 55, in parse
    plugin.parse(inventory, loader, path, cache=cache)
  File "/Users/alancoding/Documents/repos/jlaska-ansible-playbooks/inventories/user_plugins/inventory_plugins/fox.py", line 43, in parse
    ancient_mystery()
  File "/Users/alancoding/Documents/repos/jlaska-ansible-playbooks/inventories/user_plugins/inventory_plugins/fox.py", line 29, in ancient_mystery
    raise Exception('Gering-ding-ding-ding-dingeringeding')
```

So, we need to run this and verify that `ancient_mystery` is in the output.

File: inventories/user_plugins/inventory_plugins/fox.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+
+class InventoryModule(BaseInventoryPlugin):
+
+    NAME = 'cow'
```
Comment (kdelee) [inline inventories/user_plugins/inventory_plugins/fox.py]: `NAME = 'fox'` ?

**Comment by AlanCoding** [comment]
Thanks for the review, @kdelee this is ready for another look.

File: inventories/user_plugins/inventory_plugins/fox.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+
+class InventoryModule(BaseInventoryPlugin):
+
+    NAME = 'fox'
+
+    def parse(self, inventory, loader, host_list, cache=True):
+        ''' doesnt parse the inventory file, but claims it did anyway '''
+        super(InventoryModule, self).parse(inventory, loader, host_list)
+        self.inventory.add_host('fox')  # was this really necessary?
```
Comment (AlanCoding) [inline inventories/user_plugins/inventory_plugins/fox.py]: Actually, this might be useful for testing the concepts of inventory rollback.

**Comment by simfarm** [comment]
Nice job with this @AlanCoding.

**Pull request merged**